### PR TITLE
Add test to catch invalid date property

### PR DESCRIPTION
**Pull request by supasate**
Add a test in `datatype.test.js` to catch invalid date property (see strongloop/loopback#1035).

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by supasate** [comment]
@superkhau Could you help review?

File: test/datatype.test.js [hunk 1]
```diff
@@ -52,6 +52,19 @@ describe('datatypes', function() {
       done();
     });
 
+  it('should throw error when property of type Date is set to invalid date',
```
Comment (superkhau) [inline test/datatype.test.js]: `it('throws an error when property of type Date is set to an invalid value',`

File: test/datatype.test.js [hunk 2]
```diff
@@ -52,6 +52,19 @@ describe('datatypes', function() {
...
+    function(done) {
```
Comment (superkhau) [inline test/datatype.test.js]: remove done

File: test/datatype.test.js [hunk 3]
```diff
@@ -52,6 +52,19 @@ describe('datatypes', function() {
...
+      var myModel = db.define('myModel', {
+        date: { type: Date },
+      });
+
+      (function() {
+        myModel.create({ date: 'invalid' });
+      }).should.throw({ message: 'Invalid date: invalid' });
+
+      done();
```
Comment (superkhau) [inline test/datatype.test.js]: remove done

**Comment by superkhau** [comment]
I applied some feedback for you, please update it and ping me again.

**Comment by supasate** [comment]
@superkhau already fixed based on your comments.

**Comment by superkhau** [comment]
LGTM. Thanks for the contribution! ;)

**Comment by supasate** [comment]
Thank you @superkhau. I think the issue at strongloop/loopback#1035 can be closed now.

**Comment by superkhau** [comment]
@supasate Done. :+1:

**Pull request merged**